Code/GraphMol/RascalMCES/RascalResult.cpp

```diff
@@ -0,0 +1,817 @@
+  std::vector<int> mapping;
+  auto numFrags = RDKit::MolOps::getMolFrags(*mol1_frags, mapping);
+  int lfs = -1;
+  for (unsigned int i = 0; i < numFrags; ++i) {
+    auto fragSize = std::count(mapping.begin(), mapping.end(), i);
+    if (fragSize > lfs) {
+      lfs = fragSize;
+    }
+  }
+  return lfs;
+}
+
+int RascalResult::getNumFrags() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  if (d_numFrags == -1) {
+    std::unique_ptr<RDKit::ROMol> mol1_frags(makeMolFrags(1));
+    std::vector<int> mol1_frag_mapping;
+    d_numFrags = RDKit::MolOps::getMolFrags(*mol1_frags, mol1_frag_mapping);
+  }
+  return d_numFrags;
+}
+
+int RascalResult::getRingNonRingBondScore() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  if (d_ringNonRingBondScore == -1) {
+    d_ringNonRingBondScore = calcRingNonRingScore();
+  }
+  return d_ringNonRingBondScore;
+}
+
+int RascalResult::getAtomMatchScore() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  if (d_atomMatchScore == -1) {
+    d_atomMatchScore = calcAtomMatchScore();
+  }
+  return d_atomMatchScore;
+}
+
+int RascalResult::getMaxDeltaAtomAtomDist() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  if (d_maxDeltaAtomAtomDist == -1) {
+    d_maxDeltaAtomAtomDist = calcMaxDeltaAtomAtomDistScore();
+  }
+  return d_maxDeltaAtomAtomDist;
+}
+
+int RascalResult::getLargestFragSize() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  if (d_largestFragSize == -1) {
+    d_largestFragSize = calcLargestFragSize();
+  }
+  return d_largestFragSize;
+}
+
+std::string RascalResult::getSmarts() const {
+  if (!d_mol1 || !d_mol2) {
+    return "";
+  }
+  if (d_smarts.empty()) {
+    d_smarts = createSmartsString();
+  }
+  return d_smarts;
+}
+
+const std::shared_ptr<ROMol> RascalResult::getMcesMol() const {
+  if (d_mcesMol || !d_mol1) {
+    return d_mcesMol;
+  }
+
+  std::set<int> mol1Bonds;
+  for (const auto &bm : d_bondMatches) {
+    mol1Bonds.insert(bm.first);
+  }
+  std::set<int> mol1Atoms;
+  for (const auto &am : d_atomMatches) {
+    mol1Atoms.insert(am.first);
+  }
+  RWMol tmpMol(*d_mol1);
+  MolOps::KekulizeIfPossible(tmpMol);
+  tmpMol.beginBatchEdit();
+  for (auto &bond : tmpMol.bonds()) {
+    if (mol1Bonds.find(bond->getIdx()) == mol1Bonds.end()) {
+      auto bo = bond->getBondType();
+      bond->getBeginAtom()->setNumExplicitHs(
+          bond->getBeginAtom()->getNumExplicitHs() + bo);
+      bond->getEndAtom()->setNumExplicitHs(
+          bond->getEndAtom()->getNumExplicitHs() + bo);
+      tmpMol.removeBond(bond->getBeginAtomIdx(), bond->getEndAtomIdx());
+    }
+  }
+  for (auto atom : tmpMol.atoms()) {
+    if (mol1Atoms.find(atom->getIdx()) == mol1Atoms.end()) {
+      tmpMol.removeAtom(atom);
+    }
+  }
+  tmpMol.commitBatchEdit();
+  MolOps::removeHs(tmpMol);
+  MolOps::sanitizeMol(tmpMol);
+  d_mcesMol.reset(new ROMol(tmpMol));
+  return d_mcesMol;
+}
+
+bool resultCompare(const RascalResult &res1, const RascalResult &res2) {
```

Comment:
For some of the scores lower is better, for others, higher is.  Or is the suggestion always to use <, so that some would be res2.value <= res1.value?  My personal preference is that it's clearer always to have res1 first in the comparison.
The differences in the sense of comparison are why I don't think overloading operator< is appropriate.

Reply:
I just wanted to verify that all of the comparisons are in the correct direction.